src/runtime_src/core/tools/xbutil2/SubCmdQuery.cpp

```diff
@@ -87,27 +150,29 @@ int subCmdQuery(const std::vector<std::string> &_options)
   }
 
   // -- Now process the subcommand --------------------------------------------
-  XBU::verbose(XBU::format("  Card: %ld", card));
-  XBU::verbose(XBU::format("Region: %ld", region));
 
   // Report system configuration and XRT information
   XBReport::report_system_config();
   XBReport::report_xrt_info();
 
-  auto device = xrt_core::get_userpf_device(card);
-  auto iplbuf = xrt_core::query_device<std::vector<char>>(device, xrt_core::device::QR_IP_LAYOUT_RAW);
-  auto iplayout = reinterpret_cast<const ip_layout*>(iplbuf.data());
-  auto cus = xrt_core::xclbin::get_cus(iplayout);
+  pu1_query_report();
 
-  int idx = 0;
-  for (auto cu : cus) {
-    std::cout << "CU[ " << idx++ << "]: @" << std::hex << cu << std::dec << "\n";
-  }
+// It is believed that if there isn't an xclbin load a boost cast will fail.
+// Needs to be investigated more.  In the meantime, commenting out this code.
+//  auto device = xrt_core::get_userpf_device(card);
+//  auto iplbuf = xrt_core::query_device<std::vector<char>>(device, xrt_core::device::QR_IP_LAYOUT_RAW);
```

Comment:
Can you try/catch?

Reply:
Done - And cleaned up the formatting of the code as well.